src/components/Article-page.jsx

```diff
@@ -0,0 +1,40 @@
+import { useEffect, useState } from "react"
+import { useParams } from 'react-router-dom'
+import { fetchArticleById, fetchCommentsById } from "../utils/utils"
+
+function SingleArticle() {
+    const [article, setArticle] = useState([])
+    const [comments, setComments] = useState([])
+    const { id } = useParams()  
+
+    useEffect(() => {
+        fetchArticleById(id).then((articleData) => {
+            setArticle(articleData)
+        })
+    }, [id])
+
+    useEffect(() => {
+        fetchCommentsById(id).then((commentsData) => {
+            setComments(commentsData)
+        })
+    }, [])
+
```

Comment:
You could add a loading pattern here so the user knows the article is on the way if they have a slow connection ⌛ 
Here's some notes which may help https://notes.northcoders.com/courses/js-front-end/useEffect#loading-patterns